CRM/Contact/BAO/Query.php

```diff
@@ -3225,6 +3227,28 @@ public function tag(&$values) {
   }
 
   /**
+   * Where / qill clause for on_hold.
+   *
+   * @param array $values
+   */
+  public function on_hold(&$values) {
```

Comment:
restWhere() doesn't handle qill properly, so that the qill section reads, for example, "On Hold in '1, 2'". This method makes qill read "On Hold in On Hold Bounce, On Hold Opt Out".

I do wish the calculations of qill can where were separate, rather than both being done in restWhere() and its various replacements such as tag(), preferredCommunication(), etc.  But such refactoring seems far out of scope for this issue.